src/merchants/amazon_gift_card_reload.py

```diff
@@ -63,6 +68,47 @@ def web_automation(driver, merchant, amount):
         elem.send_keys(Keys.ENTER)
 
     WebDriverWait(driver, 30).until(expected_conditions.element_to_be_clickable((By.ID, 'asv-manual-reload-amount')))
+
+    if merchant.dry_run == False:
+        perform_reload(driver, merchant, amount)
+
+        if 'thank-you' not in driver.current_url:
+            return Result.unverified
+
+    # Best-effort archive the first GC reload if we're asked to
+    if merchant.merchant_specific_config['archive'] == True:
+        amazon_consistency_delay_sec = 5
+
+        LOGGER.info(f'Waiting {amazon_consistency_delay_sec} seconds before attempting to archive last order...')
+        time.sleep(amazon_consistency_delay_sec)
+
+        driver.get('https://smile.amazon.com/gp/your-account/order-history/')
+
+        first_archive_link_xpath = "(//a[contains(text(),'Amazon.com Gift Card Balance Reload')]/../../../../../../../../../../..//a[contains(text(),'Archive order')])"
+
+        try:
+            WebDriverWait(driver, 10).until(
+                expected_conditions.element_to_be_clickable((By.XPATH, first_archive_link_xpath)))
+            driver.find_element_by_xpath(first_archive_link_xpath).click()
+
+            archive_popup_link_xpath = "//input[@value='archiveOrder']"
+            WebDriverWait(driver, 10).until(
+                expected_conditions.element_to_be_clickable((By.XPATH, archive_popup_link_xpath)))
+            driver.find_element_by_xpath(archive_popup_link_xpath).click()
+
+            LOGGER.info('Archived successfully.')
+        except TimeoutException:
+            LOGGER.warning("Couldn't find an order to archive. Continuing.")
+
+    return Result.success if merchant.dry_run == False else Result.dry_run
```

Comment:
might have to be `!= True` here instead - still reporting as `dry_run` for me on real transactions

Reply:
j/k, it was handling of the result code elsewhere. fixed now.